crates/metrics/collectors/rpc_calls/src/lib.rs

```diff
@@ -73,11 +73,17 @@ impl RpcCallsMetrics {
         );
 
         metrics::histogram!(key.to_string(), duration);
-
-        self.logger.debug(
-            "set rpc requet duration seconds , methods : {} , client_version: {}, duration: {} ",
-            &["eigen-metrics-collectors-rpc-calls.set_rpc_request_duration_seconds"],
-        )
+        match (&self.logger.tracing_logger, &self.logger.noop_logger) {
+            (Some(tracing_logger), _) => tracing_logger.debug(
```

Comment:
this seems very verbose, could you abstract the logger away, so that you decide on a type of logger in the beginning and then that's the one used without the caller knowing which is the current one?

Reply:
The user will initialize their logger once in the beginning, and share it .

From 2 options : Tracing logger, or Noop Logger

Logger needs to be shared across the crates, so it can't be removed from the structs (if that 's what you meant by abstracting away the logger).
There is a way to allow user to use any of the logger by using the Logger trait , instead of EigenLogger struct , but that would mean some complexity in DevEx .
Something like this
```
pub struct EigenPerformanceMetrics<'a, L: Logger> {
    /// The performance metric is a score between 0 and 100 and each developer can define their own way of calculating the score.
    /// The score is calculated based on the performance of the AVS Node and the performance of the backing services.
    eigen_metrics: EigenPerformance,

    logger: &'a L,
}

```

Should i follow the above pattern ?

Reply:
could you make the logger an Arc<dyn Logger> and thus avoid the lifetimes issue?

Reply:
you could even typedef it

Reply:
done